src/snap/NotificationsEmitter.ts

```diff
@@ -0,0 +1,89 @@
+import { StrictEventEmitter } from "strict-event-emitter";
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+import pEvent = require("p-event");
+import { DappeteerPage } from "../page";
+import * as dappeteer from "../../src";
+import { clickOnElement, openProfileDropdown } from "../helpers";
+import { NotificationItem, NotificationList } from "./types";
+
+interface EventsMap {
+  newNotification: (notification: NotificationItem) => void;
+}
+
+class NotificationsEmitter extends StrictEventEmitter<EventsMap> {
+  private notifications: NotificationList = [];
+  private notificationsTab: DappeteerPage;
+  private readonly emitter: StrictEventEmitter<EventsMap>;
+  private readonly page: DappeteerPage;
+
+  constructor(protected metamask: dappeteer.Dappeteer) {
+    super();
+    this.page = metamask.page;
+    this.emitter = new StrictEventEmitter<EventsMap>();
+    this.configureEmitterListener();
+  }
+
+  private configureEmitterListener(): void {
+    this.emitter.on("newNotification", (notification: NotificationItem) => {
```

Comment:
this is memory leak! I don't think we need this whole method at all. I would even remove notifications array as we have separate method for that.